### ISSUE TITLE
Interface change: "strategy"

### ISSUE THREAD

**Issue by mayer79** [mayer79/kernelshap#28]
Currently, there are four sampling strategies implemented:

1. `exact = TRUE`
2. `exact = "auto"`
3. `paired_sampling = TRUE` (and exact = FALSE)
4. `paired_sampling = FALSE` (and exact = FALSE)

For the user, it is relatively difficult to see how these arguments interact. Furthermore, additional strategies might be added in the future.

A clearer interface would be to replace the two arguments "exact" and "paired_sampling" by a single one, "strategy":

1. `strategy = "auto"` (same as exact = auto)
2. `strategy = "exact"` (same as exact = TRUE)
3. `strategy = "paired_sampling"` (same as exact = FALSE with paired_sampling = TRUE)
4. `strategy = "simple_sampling"` (same as exact = FALSE with paired_sampling = FALSE)

**Comment by dswatson** [comment]
I agree it makes more sense to have a single argument take care of this. Instead of `"auto"`, I would just leave this `NULL` with default behavior kicking in if the argument is not overriden. (This is standard in most R packages, see e.g. `ranger`.) I would also add a comment in details noting that there is basically never any reason to set `sampling_stategy = "simple"`, except perhaps to test convergence rates against `sampling_strategy = "paired"`, which tends to dominate in non-exact regimes.

**Comment by mayer79** [comment]
Actually, I think it is better to use "auto" as strategy, because it is a bit cleaner with type hinting (via "match.arg"). But I will change the default for m ("auto") to NULL.

**Comment by mayer79** [comment]
Implemented in https://github.com/mayer79/kernelshap/pull/29
